libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo/interceptor/RetryOnErrorInterceptor.kt

```diff
@@ -61,39 +72,47 @@ private class DefaultRetryOnErrorInterceptorImpl(private val networkMonitor: Net
     val downStream = chain.proceed(request)
 
     return flow {
-          if (failFastIfOffline && networkMonitor?.isOnline?.value == false) {
-            emit((ApolloResponse.Builder(request.operation, request.requestUuid).exception(OfflineApolloException).build()))
-          } else {
-            emitAll(downStream)
-          }
-        }.onEach {
-          if (retryOnError && it.exception != null && it.exception!!.isRecoverable()) {
-            throw RetryException
-          } else {
-            attempt = 0
-          }
-        }.retryWhen { cause, _ ->
-          if (cause is RetryException) {
-            attempt++
-            if (networkMonitor != null) {
-              networkMonitor.waitForNetwork()
-            } else {
-              delay(2.0.pow(attempt).seconds)
-            }
-            true
-          } else {
-            // Not a RetryException, probably a programming error, pass it through
-            false
-          }
+      if (failFastIfOffline && networkMonitor?.isOnline?.value == false) {
+        emit((ApolloResponse.Builder(request.operation, request.requestUuid).exception(OfflineApolloException).build()))
+      } else {
+        emitAll(downStream)
+      }
+    }.onEach {
+      if (retryOnError && it.exception != null && isRecoverable(it.exception!!)) {
+        throw RetryException
+      } else {
+        attempt = 0
+      }
+    }.retryWhen { cause, _ ->
+      if (cause is RetryException) {
+        attempt++
+        if (networkMonitor != null) {
+          networkMonitor.waitForNetwork()
+        } else {
+          delay(2.0.pow(attempt).seconds)
         }
+        true
+      } else {
+        // Not a RetryException, probably a programming error, pass it through
+        false
+      }
+    }
   }
 }
 
 private fun ApolloException.isRecoverable(): Boolean {
   /**
    * TODO: refine this. Some networks errors are probably not recoverable (SSL errors probably, maybe others?)
    */
-  return this is ApolloNetworkException
+  return when (this) {
+    is ApolloNetworkException,
```

Comment:
IIRC `OfflineApolloException` (which is a `ApolloNetworkException`) should be excluded here, otherwise we retry when we should fail fast if offline